### PR TITLE
Replace mega hack with mini hack.

### PR DESCRIPTION
**Pull request by notlee**
Makes the aside-sidebar of the query layout optional.

I think this is better for users than the [wrapper approach](https://github.com/Financial-Times/o-layout/pull/64) and the previous [modifier approach](https://github.com/Financial-Times/o-layout/pull/65) because it requires no additional markup. And we get to remove the `_getMainMaxWidth` mega hack 🎉 

_Unrelated this PR also corrects footer markup, and updates the migration guide accordingly._

### Good
- Simplifies the SCSS (removes `_getMainMaxWidth` super grossness).
- Makes the css grid more flexible as the main content can be expressed in fractions rather than `px`.
- To not use a sidebar on the query layout, just don't include the sidebar element. 
- Easy to document and understand.

### Bad
- The grid gutter for the sidebar takes up space even when the sidebar is empty.

screenshot 2019-01-25 at 17 02 21

### CODE REVIEW

**Comment by notlee** [comment]
There's no mega hack but there is a mini hack 😅 I'll add code comments on Monday
```left: #{calc((100vw - 100%) * -0.5)};```

**Comment by notlee** [comment]
Our modivation is to support registry-like layouts and layouts like that pioneered by bizops.

screenshot 2019-01-28 at 11 56 49
screenshot 2019-01-28 at 11 57 39

File: README.md [hunk 1]
```diff
@@ -295,7 +295,7 @@ Actions overview with 4 items with footer:
 
 ## Query Layout
 
-The query layout is intended for search, filter, and result pages.The query layout provides four areas (in addition to a header and a footer):
+The query layout is intended for search, filter, and result pages. The query layout provides four areas (in addition to a header and a footer):
```
Comment (gvonkoss) [inline README.md]:
```suggestion
The query layout is intended for search, filter, and result pages. It provides four areas (in addition to a header and a footer):
```

File: README.md [hunk 2]
```diff
@@ -341,6 +341,22 @@ The query layout is intended for search, filter, and result pages.The query layo
 └————————————————————————————┘
 ```
 
+To make the main content and heading area larger on desktop, delete the aside sidebar element `o-layout__aside-sidebar`.
```
Comment (gvonkoss) [inline README.md]: Could we say something more along the line of 'this layout can be used without a right-hand side bar, which you will get by delelting the aside ....'

File: src/scss/_grid.scss [hunk 3]
```diff
@@ -122,23 +125,23 @@
...
+				"query-sidebar heading"
+				"query-sidebar main"
+				"query-sidebar aside-sidebar"
+				"footer footer";
 		};
 
 		@include oGridRespondTo($from: L) {
 			grid-template-rows: auto auto 1fr auto;
-			grid-template-columns: 1fr minmax(auto, $_o-layout-sidebar-max-width) minmax(15rem, _getMainMaxWidth($_o-layout-sidebar-max-width, $_o-layout-sidebar-max-width)) minmax(auto, $_o-layout-sidebar-max-width) 1fr;
+			grid-template-columns: $_o-layout-sidebar $_o-layout-main fit-content($_o-layout-sidebar-max-width); // fit-content makes the aside collapse with no content (optional)
```
Comment (gvonkoss) [inline src/scss/_grid.scss]: 🎉

File: src/scss/_grid.scss [hunk 4]
```diff
@@ -12,15 +12,26 @@
...
 			"footer";
 	}
+
+	.o-layout__footer,
+	.o-layout__header {
+		// Position to the left of the viewport.
+		// Relative to the center aligned grid.
+		position: relative;
+		width: 100vw;
+		left: #{calc((100vw - 100%) * -0.5)};
```
Comment (gvonkoss) [inline src/scss/_grid.scss]: So this is happening because the grid doesn't span the whole page, right? Why did we remove the five columns again?

Comment (notlee) [reply src/scss/_grid.scss]: Yeah, it's to remove the mega hack which calculated in SCSS a px value for the main content area. We either need a wrapper around the main content and sidebar, or we set the whole of `.o-layout` to the max width of the container and then make the header and footer "break out" of the container.

Comment (notlee) [reply src/scss/_grid.scss]: Ah, and the mega hack is bad because if the main content area is set in px it can't expand when the sidebar is deleted.

**Pull request merged**